Apigee/Mint/DataStructures/DataStructure.php

```diff
@@ -107,4 +108,34 @@ public function __get($name)
             return null;
         }
     }
+
+    public function __toString()
+    {
+        $data = array();
+        $reflect = new \ReflectionClass($this);
+        /** @var \ReflectionProperty $property */
+        foreach ($reflect->getProperties() as $property) {
+            $method_name = 'get' . ucfirst($property->getName());
+            $value = null;
+            if (method_exists($this, $method_name)) {
+                $value = $this->{$method_name}();
+            }
+            // Additional check for getters of boolean properties.
+            // Ex.: $this->isBoolean; $this->isBoolean();
+            elseif (method_exists($this, $property->getName())) {
+                $value = $this->{$property->getName()}();
+            }
+
+            if (is_object($value)) {
+                if (method_exists($value, '__toString')) {
```

Comment:
Ignore objects, which does not have __toString() method defined. This is better (and more secure) than just simply serializing everything that json_encode() can OOTB.